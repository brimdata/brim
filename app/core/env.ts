import {app as electronApp, remote} from "electron"
const app = electronApp || (remote && remote.app)

export default {
  get isCI() {
    return process.env.GITHUB_ACTIONS === "true"
  },
  get isIntegrationTest() {
    return process.env.BRIM_ITEST === "true"
  },
  get isTest() {
    return process.env.NODE_ENV === "test"
  },
  get isDevelopment() {
    const isEnvSet = "ELECTRON_IS_DEV" in process.env
    const getFromEnv = parseInt(process.env.ELECTRON_IS_DEV, 10) === 1
    return isEnvSet ? getFromEnv : app && !app.isPackaged
  },
  get isRelease() {
    return app.isPackaged
  },
  get isMac() {
    return process.platform === "darwin"
  },
  get isWindows() {
    return process.platform === "win32"
  },
  get isLinux() {
    return process.platform === "linux"
  }
}
