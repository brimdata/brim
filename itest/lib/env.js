/* @flow */

import path from "path"

export const isCI = (): boolean => process.env.GITHUB_ACTIONS === "true"

export const repoDir = (): string =>
  path.resolve(path.join(__dirname, "..", ".."))

export const itestDir = (): string =>
  path.join(process.env.WORKSPACE || "run", "itest")

export const testDataDir = (): string =>
  path.resolve(path.join(repoDir(), "itest", "testdata"))
