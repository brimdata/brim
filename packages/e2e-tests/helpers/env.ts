import path from "path"

export const repoDir = (): string =>
  path.resolve(path.join(__dirname, "..", "..", ".."))

export const itestDir = (): string =>
  path.join(process.env.WORKSPACE || "run", "playwright-itest")

export const testDataDir = (): string =>
  path.resolve(path.join(repoDir(), "src", "test", "shared", "data"))
