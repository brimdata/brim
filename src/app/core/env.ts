import {app as electronApp} from "electron"

const isPackaged = () =>
  electronApp
    ? electronApp.isPackaged
    : require("@electron/remote")?.app?.isPackaged

export default {
  get isCI() {
    return process.env.GITHUB_ACTIONS === "true"
  },
  get isIntegrationTest() {
    return process.env.BRIM_ITEST === "true"
  },
  get isTest() {
    return process.env.NODE_ENV === "test"
  },
  get isDevelopment() {
    const isEnvSet = "ELECTRON_IS_DEV" in process.env
    const getFromEnv = parseInt(process.env.ELECTRON_IS_DEV, 10) === 1
    return isEnvSet ? getFromEnv : isPackaged()
  },
  get isRelease() {
    return isPackaged()
  },
  get isMac() {
    return process.platform === "darwin"
  },
  get isWindows() {
    return process.platform === "win32"
  },
  get isLinux() {
    return process.platform === "linux"
  },
}
