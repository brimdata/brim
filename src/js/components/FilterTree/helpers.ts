import {isEqual} from "lodash"

import {Finding} from "../../state/Investigation/types"
import Tree from "../../models/Tree"

export function createInvestigationTree(investigation: Finding[]) {
  const tree = new Tree({data: "ROOT", children: []})

  for (const finding of investigation) {
    let node = tree.getRoot()

    eachFilter(finding.search, (filter) => {
      if (!node) return
      const nextNode = node.children.find((child) =>
        isEqual(child.data.filter, filter)
      )
      if (nextNode) {
        node = nextNode
      } else {
        node = node.addChild({filter, finding})
      }
    })
  }

  return tree
}

function eachFilter(searchBar, callback) {
  const {pins, program} = searchBar
  const filters = [...pins]

  if (!/^\s*$/.test(program)) filters.push(program)

  return filters.forEach(callback)
}
