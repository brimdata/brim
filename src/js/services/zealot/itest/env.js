/* @flow */
import path from "path"

export const ztestDir = (): string => path.join("run", "ztest")
