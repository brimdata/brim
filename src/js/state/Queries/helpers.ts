import TreeModel from "tree-model"
import {Group, Query} from "./types"
import {last} from "lodash"

export const flattenQueryTree = (root: Group, includeFolders = true) => {
  return new TreeModel({childrenPropertyName: "items"}).parse(root).all((n) => {
    return n.model.id !== "root" && (includeFolders || !("items" in n))
  })
}

export const getNextCount = (
  queries: Query[],
  type: "Session" | "Query"
): number => {
  const regex = type === "Session" ? /^Session #\d+$/ : /^Query #\d+$/
  return (
    (last(
      queries
        .filter((q) => regex.test(q.name))
        .map((q) => parseInt(q.name.split("#")[1]))
        .sort((a, b) => a - b)
    ) ?? 0) + 1
  )
}
