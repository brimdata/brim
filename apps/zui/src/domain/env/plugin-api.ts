import env from "src/app/core/env"
import {getPath} from "src/js/api/core/get-path"
import open from "src/js/lib/open"
import {join} from "path"

export class EnvApi {
  get isWindows() {
    return env.isWindows
  }

  get isMac() {
    return env.isMac
  }

  get isLinux() {
    return env.isLinux
  }

  get isTest() {
    return env.isTest
  }

  openExternal(uri: string, opts: {newWindow?: boolean} = {}) {
    return open(uri, opts)
  }

  getExePath(relPath: string) {
    return join(getPath("zdeps"), relPath) + (env.isWindows ? ".exe" : "")
  }
}
