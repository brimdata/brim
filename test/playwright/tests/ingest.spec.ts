import path from "path"
import {testDataDir} from "../helpers/env"
import TestApp from "../helpers/test-app"

describe("Ingest tests", () => {
  const app = new TestApp("Ingest tests")

  beforeAll(async () => {
    await app.init()
  })

  afterAll(async () => {
    await app.shutdown()
  })

  const searchZql =
    '_path=="conn" proto=="tcp" | cut ts, id.orig_h, id.orig_p, id.resp_h, id.resp_p, proto | sort ts'
  const sampleFiles = [
    "sample.pcap",
    "sample.pcapng",
    "sample.tsv",
    "sample.ndjson",
    "sample.zng"
  ].map((f) => [path.normalize(path.join(testDataDir(), f)), f])

  sampleFiles.forEach(([path, name]) => {
    test(`ingest of ${name}`, async () => {
      await app.ingestFiles([path])
      await app.search(searchZql)
      const results = await app.getViewerResults()
      expect(results).toMatchSnapshot(name)
    })
  })
})
