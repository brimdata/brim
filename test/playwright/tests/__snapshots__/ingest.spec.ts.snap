// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingest tests ingest of sample.ndjson: sample.ndjson 1`] = `
Array [
  "ts",
  "proto",
  "2020-02-25T16:03:05.983635Z",
  "tcp",
  "2020-02-25T16:03:06.154443Z",
  "tcp",
  "2020-02-25T16:03:07.715728Z",
  "tcp",
  "2020-02-25T16:03:07.715839Z",
  "tcp",
  "2020-02-25T16:03:08.210507Z",
  "tcp",
  "2020-02-25T16:03:08.334796Z",
  "tcp",
  "2020-02-25T16:03:08.449312Z",
  "tcp",
  "2020-02-25T16:03:08.472700Z",
  "tcp",
  "2020-02-25T16:03:08.807682Z",
  "tcp",
  "2020-02-25T16:03:10.264455Z",
  "tcp",
  "2020-02-25T16:03:10.938093Z",
  "tcp",
  "2020-02-25T16:03:12.860963Z",
  "tcp",
  "2020-02-25T16:03:13.996366Z",
  "tcp",
  "2020-02-25T16:03:15.784014Z",
  "tcp",
  "2020-02-25T16:03:15.986756Z",
  "tcp",
]
`;

exports[`Ingest tests ingest of sample.pcap: sample.pcap 1`] = `
Array [
  "ts",
  "id › orig_h",
  "id › orig_p",
  "id › resp_h",
  "id › resp_p",
  "proto",
  "2020-02-25T16:03:05.983",
  "192.168.1.110",
  "56625",
  "172.217.9.142",
  "443",
  "tcp",
  "2020-02-25T16:03:06.154",
  "192.168.1.110",
  "57591",
  "172.217.9.142",
  "443",
  "tcp",
  "2020-02-25T16:03:07.715",
  "192.168.1.110",
  "55635",
  "18.246.31.137",
  "443",
  "tcp",
  "2020-02-25T16:03:07.715",
  "192.168.1.110",
  "55747",
  "13.52.5.22",
  "443",
  "tcp",
  "2020-02-25T16:03:08.210",
  "192.168.1.110",
  "55344",
  "52.37.243.173",
  "443",
  "tcp",
  "2020-02-25T16:03:08.334",
  "192.168.1.110",
  "55346",
  "52.37.243.173",
  "443",
  "tcp",
  "2020-02-25T16:03:08.449",
  "192.168.1.110",
  "57326",
  "173.194.201.189",
  "443",
  "tcp",
  "2020-02-25T16:03:08.472",
  "192.168.1.110",
  "57487",
  "192.30.253.125",
  "443",
  "tcp",
  "2020-02-25T16:03:08.807",
  "192.168.1.110",
  "55354",
  "52.37.243.173",
  "443",
  "tcp",
  "2020-02-25T16:03:10.264",
  "192.168.1.110",
  "57332",
  "64.233.179.189",
  "443",
  "tcp",
  "2020-02-25T16:03:10.938",
  "192.168.1.110",
  "55351",
  "18.205.93.211",
  "443",
  "tcp",
  "2020-02-25T16:03:12.860",
  "192.168.1.110",
  "57635",
  "17.125.252.5",
  "443",
  "tcp",
  "2020-02-25T16:03:13.996",
  "192.168.1.110",
  "57640",
  "209.216.230.240",
  "443",
  "tcp",
  "2020-02-25T16:03:15.784",
  "192.168.1.110",
  "57572",
  "172.217.6.138",
  "443",
  "tcp",
  "2020-02-25T16:03:15.986",
  "192.168.1.110",
  "57540",
  "172.217.1.138",
  "443",
  "tcp",
]
`;

exports[`Ingest tests ingest of sample.pcapng: sample.pcapng 1`] = `
Array [
  "ts",
  "id › orig_h",
  "id › orig_p",
  "id › resp_h",
  "id › resp_p",
  "proto",
  "2020-02-25T16:03:05.983",
  "192.168.1.110",
  "56625",
  "172.217.9.142",
  "443",
  "tcp",
  "2020-02-25T16:03:06.154",
  "192.168.1.110",
  "57591",
  "172.217.9.142",
  "443",
  "tcp",
  "2020-02-25T16:03:07.715",
  "192.168.1.110",
  "55635",
  "18.246.31.137",
  "443",
  "tcp",
  "2020-02-25T16:03:07.715",
  "192.168.1.110",
  "55747",
  "13.52.5.22",
  "443",
  "tcp",
  "2020-02-25T16:03:08.210",
  "192.168.1.110",
  "55344",
  "52.37.243.173",
  "443",
  "tcp",
  "2020-02-25T16:03:08.334",
  "192.168.1.110",
  "55346",
  "52.37.243.173",
  "443",
  "tcp",
  "2020-02-25T16:03:08.449",
  "192.168.1.110",
  "57326",
  "173.194.201.189",
  "443",
  "tcp",
  "2020-02-25T16:03:08.472",
  "192.168.1.110",
  "57487",
  "192.30.253.125",
  "443",
  "tcp",
  "2020-02-25T16:03:08.807",
  "192.168.1.110",
  "55354",
  "52.37.243.173",
  "443",
  "tcp",
  "2020-02-25T16:03:10.264",
  "192.168.1.110",
  "57332",
  "64.233.179.189",
  "443",
  "tcp",
  "2020-02-25T16:03:10.938",
  "192.168.1.110",
  "55351",
  "18.205.93.211",
  "443",
  "tcp",
  "2020-02-25T16:03:12.860",
  "192.168.1.110",
  "57635",
  "17.125.252.5",
  "443",
  "tcp",
  "2020-02-25T16:03:13.996",
  "192.168.1.110",
  "57640",
  "209.216.230.240",
  "443",
  "tcp",
  "2020-02-25T16:03:15.784",
  "192.168.1.110",
  "57572",
  "172.217.6.138",
  "443",
  "tcp",
  "2020-02-25T16:03:15.986",
  "192.168.1.110",
  "57540",
  "172.217.1.138",
  "443",
  "tcp",
]
`;

exports[`Ingest tests ingest of sample.tsv: sample.tsv 1`] = `
Array [
  "ts",
  "id › orig_h",
  "id › orig_p",
  "id › resp_h",
  "id › resp_p",
  "proto",
  "2020-02-25T16:03:05.983",
  "192.168.1.110",
  "56625",
  "172.217.9.142",
  "443",
  "tcp",
  "2020-02-25T16:03:06.154",
  "192.168.1.110",
  "57591",
  "172.217.9.142",
  "443",
  "tcp",
  "2020-02-25T16:03:07.715",
  "192.168.1.110",
  "55635",
  "18.246.31.137",
  "443",
  "tcp",
  "2020-02-25T16:03:07.715",
  "192.168.1.110",
  "55747",
  "13.52.5.22",
  "443",
  "tcp",
  "2020-02-25T16:03:08.210",
  "192.168.1.110",
  "55344",
  "52.37.243.173",
  "443",
  "tcp",
  "2020-02-25T16:03:08.334",
  "192.168.1.110",
  "55346",
  "52.37.243.173",
  "443",
  "tcp",
  "2020-02-25T16:03:08.449",
  "192.168.1.110",
  "57326",
  "173.194.201.189",
  "443",
  "tcp",
  "2020-02-25T16:03:08.472",
  "192.168.1.110",
  "57487",
  "192.30.253.125",
  "443",
  "tcp",
  "2020-02-25T16:03:08.807",
  "192.168.1.110",
  "55354",
  "52.37.243.173",
  "443",
  "tcp",
  "2020-02-25T16:03:10.264",
  "192.168.1.110",
  "57332",
  "64.233.179.189",
  "443",
  "tcp",
  "2020-02-25T16:03:10.938",
  "192.168.1.110",
  "55351",
  "18.205.93.211",
  "443",
  "tcp",
  "2020-02-25T16:03:12.860",
  "192.168.1.110",
  "57635",
  "17.125.252.5",
  "443",
  "tcp",
  "2020-02-25T16:03:13.996",
  "192.168.1.110",
  "57640",
  "209.216.230.240",
  "443",
  "tcp",
  "2020-02-25T16:03:15.784",
  "192.168.1.110",
  "57572",
  "172.217.6.138",
  "443",
  "tcp",
  "2020-02-25T16:03:15.986",
  "192.168.1.110",
  "57540",
  "172.217.1.138",
  "443",
  "tcp",
]
`;

exports[`Ingest tests ingest of sample.zng: sample.zng 1`] = `
Array [
  "ts",
  "id › orig_h",
  "id › orig_p",
  "id › resp_h",
  "id › resp_p",
  "proto",
  "2020-02-25T16:03:05.983",
  "192.168.1.110",
  "56625",
  "172.217.9.142",
  "443",
  "tcp",
  "2020-02-25T16:03:06.154",
  "192.168.1.110",
  "57591",
  "172.217.9.142",
  "443",
  "tcp",
  "2020-02-25T16:03:07.715",
  "192.168.1.110",
  "55635",
  "18.246.31.137",
  "443",
  "tcp",
  "2020-02-25T16:03:07.715",
  "192.168.1.110",
  "55747",
  "13.52.5.22",
  "443",
  "tcp",
  "2020-02-25T16:03:08.210",
  "192.168.1.110",
  "55344",
  "52.37.243.173",
  "443",
  "tcp",
  "2020-02-25T16:03:08.334",
  "192.168.1.110",
  "55346",
  "52.37.243.173",
  "443",
  "tcp",
  "2020-02-25T16:03:08.449",
  "192.168.1.110",
  "57326",
  "173.194.201.189",
  "443",
  "tcp",
  "2020-02-25T16:03:08.472",
  "192.168.1.110",
  "57487",
  "192.30.253.125",
  "443",
  "tcp",
  "2020-02-25T16:03:08.807",
  "192.168.1.110",
  "55354",
  "52.37.243.173",
  "443",
  "tcp",
  "2020-02-25T16:03:10.264",
  "192.168.1.110",
  "57332",
  "64.233.179.189",
  "443",
  "tcp",
  "2020-02-25T16:03:10.938",
  "192.168.1.110",
  "55351",
  "18.205.93.211",
  "443",
  "tcp",
  "2020-02-25T16:03:12.860",
  "192.168.1.110",
  "57635",
  "17.125.252.5",
  "443",
  "tcp",
  "2020-02-25T16:03:13.996",
  "192.168.1.110",
  "57640",
  "209.216.230.240",
  "443",
  "tcp",
  "2020-02-25T16:03:15.784",
  "192.168.1.110",
  "57572",
  "172.217.6.138",
  "443",
  "tcp",
  "2020-02-25T16:03:15.986",
  "192.168.1.110",
  "57540",
  "172.217.1.138",
  "443",
  "tcp",
]
`;
